pilot/pkg/proxy/envoy/v2/eds.go

```diff
@@ -491,17 +491,19 @@ func (s *DiscoveryServer) WorkloadUpdate(id string, labels map[string]string, an
 // on each step: instead the conversion happens once, when an endpoint is first discovered.
 func (s *DiscoveryServer) EDSUpdate(shard, serviceName string,
 	entries []*model.IstioEndpoint) error {
-	return s.edsUpdate(shard, serviceName, entries, false)
```

Comment:
Given the `edsUpdate` never returns error, we can even change `EDSUpdate` signature and then kube/controller's code [1]can be simplified as well. @costinm Let me know if this makes sense.

[1] https://github.com/istio/istio/blob/master/pilot/pkg/serviceregistry/kube/controller.go#L792